fmf/_compat_jsonschema.py

```diff
@@ -0,0 +1,91 @@
+"""Compatibility layer for jsonschema validation."""
+
+from __future__ import annotations
+
+import warnings
+from typing import Any, Dict, List, NamedTuple, Optional, Union
+
+import jsonschema
+
+try:
+    from jsonschema import validators
+    from referencing import Registry, create
+    from referencing.jsonschema import DRAFT4
+    MODERN_JSONSCHEMA = True
+except ImportError:
+    MODERN_JSONSCHEMA = False
+
+
+class JsonSchemaValidationResult(NamedTuple):
+    """Represents JSON Schema validation result."""
+    result: bool
+    errors: List[Any]
+
+
+def create_validator(
+        schema: Dict[str, Any],
+        schema_store: Optional[Dict[str, Any]] = None
+        ) -> Union[jsonschema.validators.Validator, jsonschema.validators.Draft4Validator]:
+    """Create a validator instance based on available jsonschema version."""
```

Comment:
I am more used to `create` naming convention being a static/classmethod. Just a thought if it should be `get` or `create` or something else.

Reply:
renamed